server/log_operation_manager.go

```diff
@@ -224,43 +228,53 @@ func (l *LogOperationManager) masterFor(ctx context.Context, allIDs []int64) ([]
 
 	held := l.tracker.Held()
 	heldIDs := make([]int64, 0, len(allIDs))
+	sort.Strings(allStringIDs)
 	for _, s := range held {
+		// Skip the log if it is not present in allIDs.
+		if i := sort.SearchStrings(allStringIDs, s); i >= len(allStringIDs) || allStringIDs[i] != s {
+			continue
+		}
 		id, err := strconv.ParseInt(s, 10, 64)
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse logID %v as int64", s)
 		}
 		heldIDs = append(heldIDs, id)
 	}
 
-	glog.V(1).Infof("acting as master for %d / %d: %s", len(held), len(allIDs), l.tracker)
 	return heldIDs, nil
 }
 
-func (l *LogOperationManager) updateHeldIDs(ctx context.Context, logIDs, allIDs []int64) {
+// updateHeldIDs updates the process status with the number/list of logs that
+// the instance holds mastership for.
+func (l *LogOperationManager) updateHeldIDs(ctx context.Context, logIDs, activeIDs []int64) {
 	l.heldMutex.Lock()
 	defer l.heldMutex.Unlock()
+	heldInfo := l.heldInfo(ctx, logIDs)
+	msg := fmt.Sprintf("Acting as master for %d / %d active logs: %s", len(logIDs), len(activeIDs), heldInfo)
 	if !reflect.DeepEqual(logIDs, l.lastHeld) {
 		l.lastHeld = make([]int64, len(logIDs))
 		copy(l.lastHeld, logIDs)
-		heldInfo := l.heldInfo(ctx, logIDs)
-		glog.Infof("now acting as master for %d / %d, %s", len(logIDs), len(allIDs), heldInfo)
+		glog.Info(msg)
 		if l.info.Registry.SetProcessStatus != nil {
 			l.info.Registry.SetProcessStatus(heldInfo)
 		}
 	}
+	glog.V(1).Info(msg)
 }
 
 func (l *LogOperationManager) getLogsAndExecutePass(ctx context.Context) error {
-	allIDs, err := l.getLogIDs(ctx)
+	activeIDs, err := l.getActiveLogIDs(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to retrieve full list of log IDs: %v", err)
+		return fmt.Errorf("failed to list active log IDs: %v", err)
 	}
-	logIDs, err := l.masterFor(ctx, allIDs)
+	// Skip IDs of the logs that are not active (for example, deleted or FROZEN
+	// ones). Note that we might still hold mastership for some of them.
+	// TODO(pavelkalinnikov): Resign mastership for the inactive logs.
+	logIDs, err := l.masterFor(ctx, activeIDs)
```

Comment:
The comment describes the secondary function of the call to `masterFor` rather than its primary purpose. Maybe `// Find the logs we are master for, skipping those logs that are not active...` or somesuch